packages/app/src/ui/components/VerifiableCredential/Display/VerifiableCredentialDisplay.tsx

```diff
@@ -0,0 +1,110 @@
+import CheckIcon from "@mui/icons-material/Check";
+import EditIcon from "@mui/icons-material/Edit";
+import IconButton from "@mui/material/IconButton";
+import TextField from "@mui/material/TextField";
+import Typography from "@mui/material/Typography";
+
+import { CryptkeeperVerifiableCredential } from "@src/types";
+
+import { useVerifiableCredentialDisplay } from "./useVerifiableCredentialDisplay";
+
+export interface VerifiableCredentialDisplayProps {
+  cryptkeeperVerifiableCredential: CryptkeeperVerifiableCredential;
+  onRenameVerifiableCredential: (name: string) => void;
+}
+
+export const VerifiableCredentialDisplay = ({
+  cryptkeeperVerifiableCredential,
+  onRenameVerifiableCredential,
+}: VerifiableCredentialDisplayProps): JSX.Element => {
+  const initialName = cryptkeeperVerifiableCredential.metadata.name;
+  const { isRenaming, name, register, onSubmit, onToggleRenaming } = useVerifiableCredentialDisplay({
+    initialName,
+    onRename: onRenameVerifiableCredential,
+  });
+
+  const { verifiableCredential } = cryptkeeperVerifiableCredential;
+  const issuerId =
+    typeof verifiableCredential.issuer === "string" ? verifiableCredential.issuer : verifiableCredential.issuer.id;
+
+  return (
+    <div>
+      {isRenaming ? (
+        <form onSubmit={onSubmit}>
+          <TextField
+            {...register("name")}
+            autoFocus
+            data-testid="verifiable-credential-display-rename-input"
+            id="verifiable-credential-display-rename-input"
+            size="small"
+            type="text"
+            variant="outlined"
+            onBlur={onToggleRenaming}
+          />
+
+          <IconButton
+            className="verifiable-credential-display__select-icon-button--selected mr-2"
+            data-testid="verifiable-credential-display-submit-rename"
+            size="medium"
+            type="submit"
+          >
+            <CheckIcon
+              className="verifiable-credential-display__select-icon--selected"
+              color="primary"
+              fontSize="inherit"
+            />
+          </IconButton>
+        </form>
+      ) : (
+        <div className="flex flex-row items-center text-lg font-semibold">
+          <IconButton
+            className="verifiable-credential-display__menu-icon-button"
+            data-testid="verifiable-credential-display-toggle-rename"
+            size="small"
+            onClick={onToggleRenaming}
+          >
+            <EditIcon className="verifiable-credential-display__menu-icon" color="primary" fontSize="inherit" />
+          </IconButton>
+
+          {`${name}`}
+        </div>
+      )}
+
+      <Typography variant="body1">
+        <u>Type:</u>
+
+        {` ${verifiableCredential.type.join(", ")}`}
```

Comment:
Let's get rid of spaces in front of values. This style should be done with styles not with markup.